components/grid/templates/no-data-template.md

```diff
@@ -0,0 +1,74 @@
+---
+title: No Data Template
+page_title: Grid - No Data Template
+description: Use custom no data templates in Grid for Blazor.
+slug: grid-templates-no-data
+tags: telerik,blazor,grid,templates,no,data
+published: True
+position: 38
+---
+
+
+# No Data Template
+
+The `NoDataTemplate` allows you to define custom content when the Grid has no data to show. It overrides the currently available **"No records available."**.
+
+>caption The result from the code snippet below
+
+![](images/grid-no-data-template.gif)
+
+>caption Use the NoDataTemplate to override the message shown while the data is still loading or when you try to filter a non-existing item
+
+````CSHTML
+@* The example showcases a Grid with a delay upon initializing its data and filtering on non-existing data *@
+
+<TelerikGrid Data="@MyData" Height="400px"
+             Pageable="true"
+             FilterMode="Telerik.Blazor.GridFilterMode.FilterRow">
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Id))" Width="120px" />
+        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" />
+        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
+        <GridColumn Field="@(nameof(SampleData.HireDate))" Title="Hire Date" />
+    </GridColumns>
+    <NoDataTemplate>
+        <strong>No Data available / The data is still loading...</strong>
```

Comment:
Consider adding a button here that adds a dummy data item to show people how the no-data-template works and goes away when there is data. You'd probably want to use an observable collection then.

Reply:
Done